qiskit/algorithms/time_evolvers/evolution_problem.py

```diff
@@ -0,0 +1,109 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Evolution problem class."""
+
+from typing import Union, Optional, Dict
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import Parameter
+from qiskit.opflow import OperatorBase, StateFn
+from ..list_or_dict import ListOrDict
+
+
+class EvolutionProblem:
+    """Evolution problem class.
+
+    This class is the input to time evolution algorithms and must contain information on the total
+    evolution time, a quantum state to be evolved and under which Hamiltonian the state is evolved.
+    """
+
+    def __init__(
+        self,
+        hamiltonian: OperatorBase,
+        time: float,
+        initial_state: Optional[Union[StateFn, QuantumCircuit]] = None,
+        aux_operators: Optional[ListOrDict[OperatorBase]] = None,
+        truncation_threshold: float = 1e-12,
+        t_param: Optional[Parameter] = None,
+        param_value_dict: Optional[Dict[Parameter, complex]] = None,
+    ):
+        """
+        Args:
+            hamiltonian: The Hamiltonian under which to evolve the system.
+            time: Total time of evolution.
+            initial_state: The quantum state to be evolved for methods like Trotterization.
+                For variational time evolutions, where the evolution happens in an ansatz,
+                this argument is not required.
+            aux_operators: Optional list of auxiliary operators to be evaluated with the
+                evolved ``initial_state`` and their expectation values returned.
+            truncation_threshold: Defines a threshold under which values can be assumed to be 0.
+                Used when ``aux_operators`` is provided.
+            t_param: Time parameter in case of a time-dependent Hamiltonian. This
+                free parameter must be within the ``hamiltonian``.
+            param_value_dict: Maps free parameters in the problem to values. Depending on the
+                algorithm, it might refer to e.g. a Hamiltonian or an initial state.
```

Comment:
Does it still refer to an initial state as we moved the initial point to the initializer of VarQTE? Also, as side note, I think parameterized Hamiltonians are not yet support in quantum info, only once Hamamura-san's PR on parameterized SparsePauliOps is merged we do 🙂